### PR TITLE
Add tenant() to Azure.Provisioning

### PR DESCRIPTION
**Pull request by tg-msft**

### CODE REVIEW

File: sdk/provisioning/Azure.Provisioning/src/Expressions/BicepFunction.cs [hunk 1]
```diff
@@ -9,71 +9,304 @@
...
+    /// <summary>
+    /// Creates a deterministic hash string based on the <paramref name="values"/>
+    /// provided as parameters.  This represents the <c>uniqueString</c> Bicep
+    /// function.
+    /// </summary>
+    /// <param name="values">
+    /// The value used in the hash function to create a unique string.  At
+    /// least one value is required.
+    /// </param>
+    /// <returns>A string containing 13 characters.</returns>
```
Comment (eerhardt) [inline sdk/provisioning/Azure.Provisioning/src/Expressions/BicepFunction.cs:29]: It's guaranteed to be 13?

File: sdk/provisioning/Azure.Provisioning/src/Expressions/BicepFunction.cs [hunk 2]
```diff
@@ -9,71 +9,304 @@
...
+    /// example, <c>CreateGuid("Hello", "World")</c> and
+    /// <c>CreateGuid("World", "Hello")</c> don't return the same value.
+    /// </para>
+    /// <para>
+    /// See the
+    /// <see href="https://learn.microsoft.com/azure/azure-resource-manager/bicep/bicep-functions-string#guid">
+    /// Bicep Functions Reference</see> for more.
+    /// </para>
+    /// </remarks>
+    public static BicepValue<string> CreateGuid(params BicepValue<string>[] values)
```
[the remark this answers is not in the document]
Comment (tg-msft) [reply sdk/provisioning/Azure.Provisioning/src/Expressions/BicepFunction.cs:110]: Yeah, .NET method names need a verb.

[the remark this answers is not in the document]
Comment (tg-msft) [reply sdk/provisioning/Azure.Provisioning/src/Expressions/BicepFunction.cs:110]: (Same reason we have `GetResourceGroup()` instead of `ResourceGroup()`)

**Review by eerhardt — APPROVED** [review]
Thanks!

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**